# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/android/TraversalRule.cpp
# Commit: 11f08e7b4bcf
# Full Hash: 11f08e7b4bcf06e7202e15f0ec3a3be50c906b04
# Author: James Teh <jteh@mozilla.com>
# Date: 2019-11-04 16:12:30
# Description:
#   Bug 1590929: a11y::TraversalRule::Match: Don't assume that all Accessibles have a frame. r=MarcoZ
#   
#   I don't have a test case for this crash, but the stack suggests the frame is null.
#   This can certainly happen for display: contents.
#   
# ==============================================================================

diff -r 61de50c75320 -r 11f08e7b4bcf accessible/android/TraversalRule.cpp
--- a/accessible/android/TraversalRule.cpp	Mon Nov 04 06:38:37 2019 +0000
+++ b/accessible/android/TraversalRule.cpp	Mon Nov 04 07:04:40 2019 +0000
@@ -39,7 +39,7 @@
 
   if ((state & states::OPAQUE1) == 0) {
     nsIFrame* frame = aAccessible->GetFrame();
-    if (frame->StyleEffects()->mOpacity == 0.0f) {
+    if (frame && frame->StyleEffects()->mOpacity == 0.0f) {
       return result | nsIAccessibleTraversalRule::FILTER_IGNORE_SUBTREE;
     }
   }
